ocaml/lambda/translcore.ml

```diff
@@ -58,6 +53,28 @@ let sort_must_not_be_void loc ty sort =
 
 let layout_exp sort e = layout e.exp_env e.exp_loc sort e.exp_type
 
+(* This is `Lambda.must_be_value` for the special case of record fields, where
+   we allow the unboxed float layout and pretend it's a normal `Pvalue
+   Pfloatval`.  The compiler already has legacy support for that using
+   (incorrect) value kind `Pfloatval` for their fields.
+
+   CR layouts v5: eliminate this hack.
+*)
+let record_field_kind l =
+  match l with
+  | Punboxed_float -> Pfloatval
+  | _ -> must_be_value l
+
+let check_record_field_sort loc sort repres =
```

Comment:
As with the similar function in `matching`, this is just where these checks have always been.  I suspect we can't quite do them in `store_label`, at least without some extra machinery, because these checks would result in incorrect defaulting when building the `temp_env` for mutually recursive type declaration checking.  But I'm content to delete them if you think now is the moment.